homeassistant/components/unifiprotect/data.py

```diff
@@ -231,41 +230,49 @@ def _async_update_device(
 
     @callback
     def _async_process_ws_message(self, message: WSSubscriptionMessage) -> None:
-        if message.new_obj is None:
+        """Process a message from the websocket."""
+        if (new_obj := message.new_obj) is None:
             if isinstance(message.old_obj, ProtectAdoptableDeviceModel):
                 self._async_remove_device(message.old_obj)
             return
 
-        obj = message.new_obj
-        if isinstance(obj, (ProtectAdoptableDeviceModel, NVR)):
-            if message.old_obj is None and isinstance(obj, ProtectAdoptableDeviceModel):
-                self._async_add_device(obj)
-            elif getattr(obj, "is_adopted_by_us", True):
-                self._async_update_device(obj, message.changed_data)
-
-        # trigger updates for camera that the event references
-        elif isinstance(obj, Event):
+        model_type = new_obj.model
+        if model_type is ModelType.EVENT:
+            if TYPE_CHECKING:
+                assert isinstance(new_obj, Event)
             if _LOGGER.isEnabledFor(logging.DEBUG):
-                log_event(obj)
-            if obj.type is EventType.DEVICE_ADOPTED:
-                if obj.metadata is not None and obj.metadata.device_id is not None:
-                    device = self.api.bootstrap.get_device_from_id(
-                        obj.metadata.device_id
-                    )
-                    if device is not None:
-                        self._async_add_device(device)
-            elif obj.camera is not None:
-                self._async_signal_device_update(obj.camera)
-            elif obj.light is not None:
-                self._async_signal_device_update(obj.light)
-            elif obj.sensor is not None:
-                self._async_signal_device_update(obj.sensor)
-        # alert user viewport needs restart so voice clients can get new options
-        elif len(self.api.bootstrap.viewers) > 0 and isinstance(obj, Liveview):
+                log_event(new_obj)
+            if (
+                (new_obj.type is EventType.DEVICE_ADOPTED)
+                and (metadata := new_obj.metadata)
+                and (device_id := metadata.device_id)
+                and (device := self.api.bootstrap.get_device_from_id(device_id))
+            ):
+                self._async_add_device(device)
+            elif camera := new_obj.camera:
+                self._async_signal_device_update(camera)
```

Comment:
Missing coverage on this line is existing.  I don't think its reachable, but I'm not ready to make that call yet

Reply:
Its reachable, its a problem with the mocking

Reply:
Fixing it is more involved so I've tested it manually instead